samples/mqtt/sender/main.go

```diff
@@ -29,18 +29,17 @@ func main() {
 	if err != nil {
 		log.Fatalf("failed to connect to mqtt broker: %s", err.Error())
 	}
-	clientConfig := &paho.ClientConfig{
-		Conn: conn,
+	config := &paho.ClientConfig{
+		ClientID: "sender-client-id",
+		Conn:     conn,
 	}
-	connConfig := &paho.Connect{
-		ClientID:   "sender-client-id",
+	// optional connect option
+	connOpt := &paho.Connect{
 		KeepAlive:  30,
 		CleanStart: true,
 	}
 	// set a default topic with test-topic1
-	p, err := mqtt_paho.New(ctx, clientConfig, connConfig, &paho.Publish{
-		Topic: "test-topic1", QoS: 0,
-	}, nil)
+	p, err := mqtt_paho.New(config, mqtt_paho.WithPublish(&paho.Publish{Topic: "test-topic1"}), mqtt_paho.WithConnect(connOpt))
 	if err != nil {
 		log.Fatalf("failed to create protocol: %s", err.Error())
```

Comment:
nit: you can use `%v` and just `err` to reduce typing :)

Reply:
Done.